### PR TITLE
refactor(transactions) Switch Payment to new transaction scheme

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

This PR primarily moves Payment to the new transaction schema, hooks, typescript, and more tests. #294

Additional Improvements
- Added ability to specify i18n instance to createWrapper factories. This will allow the english config to be used.
- Moved types for `IssuedCurrency`, `IssuedCurrencyAmount`, `Amount` and `ExplorerAmount` to their own file. `shared/types.ts`.
- Used `SimpleRow` and `Amount` components
- Added 100% code coverage to the payment transaction.
- Moved location of "partial payment allowed" to be within the amount row.  For currency conversions it would show outside.
- Updated the text for SendMax when it a currency conversion to "Convert Max" to match what TableDetail showed.  It previously would show "Using at most"

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [x] Updated files to TypeScript

## Before 
Screen Shot 2022-10-03 at 2 45 08 PM
Screen Shot 2022-10-03 at 3 56 55 PM
Screen Shot 2022-10-03 at 4 00 01 PM

## After
Screen Shot 2022-10-03 at 2 56 06 PM
Screen Shot 2022-10-03 at 3 57 36 PM
Screen Shot 2022-10-03 at 4 00 01 PM

## Test Plan

These are the transactions to compare against.

### Standard Payment
[F4AD54E6FE56B53F8FB8B422C4E9636D6AD56682EA45FF3C8AFBC66EF2D82612](https://livenet.xrpl.org/transactions/F4AD54E6FE56B53F8FB8B422C4E9636D6AD56682EA45FF3C8AFBC66EF2D82612)

### Failed Conversion Payment
[DA9E41A957D5683DA87B830057E98C755BCFDD247892A7E96603A68C6ED416BA](https://livenet.xrpl.org/transactions/DA9E41A957D5683DA87B830057E98C755BCFDD247892A7E96603A68C6ED416BA)

### Partial Payment
[439B1D82C4739E4B3DD1FE440AE0EBE62E30C64FA7FF7DC0423B95CAF2FEB384](https://livenet.xrpl.org/transactions/439B1D82C4739E4B3DD1FE440AE0EBE62E30C64FA7FF7DC0423B95CAF2FEB384)

### Source and Destination Tag
[6E0EC20AE77ED25731B21B5B0B8FF7EA36605CBAA9C7D77725B6EC4D43E5C009](https://livenet.xrpl.org/transactions/6E0EC20AE77ED25731B21B5B0B8FF7EA36605CBAA9C7D77725B6EC4D43E5C009)

### SendMax
[AABF5991C4540CB1DF7D4F5DC8FEA8AC1F2DB4ECB7BFEBC96B8752CECF482325](https://livenet.xrpl.org/transactions/AABF5991C4540CB1DF7D4F5DC8FEA8AC1F2DB4ECB7BFEBC96B8752CECF482325)

### CODE REVIEW

File: src/containers/shared/components/Amount.tsx [hunk 1]
```diff
@@ -3,15 +3,10 @@ import { CURRENCY_OPTIONS, XRP_BASE } from '../transactionUtils'
...
-  value:
-    | string
-    | {
-        issuer?: string
-        currency: string
-        amount: number
-      }
+  value: ExplorerAmount
   displayIssuer?: boolean
 }
```
Comment (mvadari) [inline src/containers/shared/components/Amount.tsx:11]: Is the amount object going to link to the token instead of the issuer?

**Review by mvadari — COMMENTED** [review]
Can you add before/after screenshots of DetailedTab and Description as well?

File: src/rippled/lib/txSummary/index.js [hunk 2]
```diff
@@ -39,7 +37,7 @@ const getInstructions = (tx, meta) => {
   return mappingFn ? mappingFn(tx, meta) : {}
 }
 
-const summarizeTransaction = (d, details = false) => ({
+export const summarizeTransaction = (d, details = false) => ({
```
Comment (mvadari) [inline src/rippled/lib/txSummary/index.js:40]: This is already exported below, on line 55 - it doesn't also need to be exported here.

File: src/rippled/lib/utils.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { convertRippleDate, EPOCH_OFFSET } from './convertRippleDate'
-import summarizeTransaction from './txSummary'
+import { summarizeTransaction } from './txSummary'
```
Comment (mvadari) [inline src/rippled/lib/utils.js:2]: Why was this changed?

File: src/containers/shared/components/Amount.tsx (hunk 1 above)
Comment (ckniffen) [reply src/containers/shared/components/Amount.tsx:11]: That is on the road map.  It will be a change to the Currency component.

File: src/rippled/lib/txSummary/index.js (hunk 2 above)
Comment (ckniffen) [reply src/rippled/lib/txSummary/index.js:40]: I was having issues building until I changed this.  I don't know what suddenly caused it.

**Comment by ckniffen** [comment]
> Can you add before/after screenshots of DetailedTab and Description as well?

I have added them now.

File: src/containers/shared/components/Amount.tsx (hunk 1 above)
Comment (ckniffen) [reply src/containers/shared/components/Amount.tsx:11]: Went ahead and created the PR #384

**Comment by JST5000** [comment]
Is `ConvertMax` a term used in transactions somewhere? I personally feel it's a bit confusing as to what it means. My preferred method would be to use the following wording and flip the top and bottom rows to make it more human readable. 

"Converting up to ________  Issued Currency"
"By using at most ________ XRP"

(Basically similar wording to the Description)

But I don't know if that violates a convention we have on the site.

**Comment by ckniffen** [comment]
> Is `ConvertMax` a term used in transactions somewhere? I personally feel it's a bit confusing as to what it means. My preferred method would be to use the following wording and flip the top and bottom rows to make it more human readable.
> 
> "Converting up to ________ Issued Currency" "By using at most ________ XRP"
> 
> (Basically similar wording to the Description)
> 
> But I don't know if that violates a convention we have on the site.

Covert Max is actually using the value of send max.  The order of it on the tableDetail is the same as the simple view.  @mduo13 which order do you think makes the most sense?

**Review by jonathanlei — APPROVED** [review]
LGTM

File: public/locales/en-US/translations.json [hunk 4]
```diff
@@ -222,8 +222,8 @@
   "offer_lack_of_funds": "The offer was partially filled, then cancelled due to lack of funds",
   "transaction_sequence": "The transaction's sequence number is",
   "trust_set_description": "It establishes <1><0>{{amount}}</0></1> as the maximum amount of <3><0>{{currency}}</0></3> from <5><0>{{issuer}}</0></5> that <7><0>{{account}}</0></7> is willing to hold",
-  "payment_desc_line_1": "The payment is from <1><0>{{account}}</0></1> to <3><0>{{destination}}</0></3>",
-  "the_source_tag_is": "The source tag is",
+  "payment_desc_line_1": "The payment is from <source/> to <destination/>",
+  "the_source_tag_is": "The source tag is ",
   "the_destination_tag_is": "The destination tag is",
   "payment_desc_line_4": "It was instructed to deliver",
```
Comment (JST5000) [inline public/locales/en-US/translations.json:228]: Is there a reason there's no `payment_desc_line_2` and line 3?

File: src/containers/shared/components/Transaction/Payment/Simple.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+          {renderPartial()}
+        </SimpleRow>
+        {sourceTag !== undefined && (
+          <SimpleRow label={t('source_tag')} data-test="source-tag">
+            {sourceTag}
+          </SimpleRow>
+        )}
+        <SimpleRow label={t('destination')} data-test="destination">
+          <Account account={parts[0]} />
+          {parts[1] && <span className="dt">:{parts[1]}</span>}
```
Comment (JST5000) [inline src/containers/shared/components/Transaction/Payment/Simple.tsx:46]: Is there ever a situation where there is no `:` to split, and so there is no parts[1]?

Comment (JST5000) [inline src/containers/shared/components/Transaction/Payment/Simple.tsx:46]: Does parts[1] cause an out of bounds error if there is no parts[1]?

Comment (JST5000) [reply src/containers/shared/components/Transaction/Payment/Simple.tsx:46]: (Or is there always a `:` in the `destination` input to be split?

Comment (ckniffen) [reply src/containers/shared/components/Transaction/Payment/Simple.tsx:46]: > Does parts[1] cause an out of bounds error if there is no parts[1]?

It will be undefined.

File: src/containers/shared/types.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  issuer: string
+}
+
+export interface IssuedCurrencyAmount extends IssuedCurrency {
+  value: string
+}
+
+export type Amount = IssuedCurrencyAmount | string
+
+export type ExplorerAmount =
```
Comment (JST5000) [inline src/containers/shared/types.ts:12]: How is `ExplorerAmount` different than `Amount`?

File: public/locales/en-US/translations.json (hunk 4 above)
Comment (ckniffen) [reply public/locales/en-US/translations.json:228]: No idea.

Comment (JST5000) [reply public/locales/en-US/translations.json:228]: Seems like in the code there are two other lines between them - so I guess those are the "lines 2" and "3" :P - Not important

File: src/containers/shared/types.ts (hunk 6 above)
Comment (ckniffen) [reply src/containers/shared/types.ts:12]: ExplorerAmount has been converted to a number

**Review by JST5000 — APPROVED** [review]
LGTM
